src/core/lib/iomgr/tcp_windows.cc

```diff
@@ -482,7 +482,10 @@ static grpc_resource_user* win_get_resource_user(grpc_endpoint* ep) {
   return tcp->resource_user;
 }
 
-static int win_get_fd(grpc_endpoint* ep) { return -1; }
+static int win_get_fd(grpc_endpoint* ep) {
+  grpc_tcp* tcp = reinterpret_cast<grpc_tcp*>(ep);
```

Comment:
This might work but I'm not certain it makes sense to implement this interface on Windows.

For example this `grpc_endpoint` `get_fd` method is supposed to return an fd as an `int`, with -1 to represent "invalid".

On Windows, though, this field being returned here is a `SOCKET`, which is an unsigned data type having the `INVALID_SOCKET` macro to represent "invalid".

I think it might be better to put the platform-specific at a higher level, e.g. extracting the chunk of code from `local_check_peer` that deals with OS handles and creating a Windows-specific version of it.

Also, we should have some test that proves this works - @yihuazhang is there a simple client/server local credentials test that is perhaps currently skipped on Windows, which could just be enabled to show that it works?

Reply:
+1 for adding a Windows version of `local_check_peer`. I will look into it. 

In C-core, we have an E2E test for local credentials in `test/core/end2end/fixtures/`  (i.e., `h2_local_{uds, ipv4, ipv6}`) but I do not how to enable these tests on Windows. Do you have any clue?  @jtattermusch, @apolcyn

Reply:
@yihuazhang I think in order to enable on Windows, those test configurations need to add "windows" to the `_platforms` field, e.g. in https://github.com/grpc/grpc/blob/ff8ceb700e8a53ed4087edc006830da372b1199a/test/core/end2end/generate_tests.bzl#L97

Reply:
Thanks @apolcyn. I should not have missed it :)